windows/src/desktop/setup/Keyman.Setup.System.SetupUILanguageManager.pas

```diff
@@ -18,6 +19,8 @@   TSetupUILanguageManager = class(TUILanguageManager)
     FStrings: TDictionary<string, TLocaleArray>;
     FActiveLocale: string;
     FLocales: TSetupLocales;
+    FIndexMapping: TDictionary<Integer,Integer>;
+    FSortedValues: TList< TPair<string,string>>;
```

Comment:
I don't think we necessarily need these as class members. We can sort `FLocales` on first external reference, and keep any sort-related variables internal to the `Reorder` function. We could make this robust by adding a check in `Locales()`, something like:

```
class function Locales: TSetupLocales;
begin
  if FHasChanged then 
  begin
    Reorder;
    FHasChanged := False;
  end;
   Result := FLocales;
end;
```

and in `RegisterSetupStrings()` add:

```
  FHasChanged := True;
```

Reply:
As I understand it `FLocales` which is of type TDictionary<string,string>; can not be ordered. I think we can create a Type that can use the `Reorder` function internally using the IndexMapping etc, and `FLocales` is declared of that type. We would then not be able to access the Dictionary directly but via public functions that could return the sorted values as an array.  Then when we "index" it, internally we would have to have the FIndexMapping look after accessing the correct `key,value` pair. 

Not sure if I am explaining what I mean well.